realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/SyncSessionTests.java

```diff
@@ -106,4 +109,52 @@ public void uploadDownloadAllChanges() throws InterruptedException {
         assertEquals(1, adminRealm.where(AllTypes.class).count());
         adminRealm.close();
     }
+
+    // check that logging out a SyncUser used by different sessions will
+    // all corresponding Realms are closed
+    @Test
+    public void logout_sameSyncUserMultipleSessions() {
+        SyncUser user = UserFactory.createUniqueUser(Constants.AUTH_URL);
+
+        SyncConfiguration syncConfiguration1 = configFactory
+                .createSyncConfigurationBuilder(user, Constants.SYNC_SERVER_URL)
+                .build();
+        Realm realm1 = Realm.getInstance(syncConfiguration1);
+
+        SyncConfiguration syncConfiguration2 = configFactory
+                .createSyncConfigurationBuilder(user, Constants.SYNC_SERVER_URL_2)
+                .build();
+        Realm realm2 = Realm.getInstance(syncConfiguration2);
+
+        SyncSession session1 = SyncManager.getSession(syncConfiguration1);
+        SyncSession session2 = SyncManager.getSession(syncConfiguration2);
+
+        // make sure the `access_token` is acquired. otherwise we can still be
+        // in WAITING_FOR_ACCESS_TOKEN state
+        SystemClock.sleep(TimeUnit.SECONDS.toMillis(2));
```

Comment:
Yeah... this is how flaky tests are made.  Is there any way to set a CountdownLatch to let the proceed?